### ISSUE TITLE
Output n_class is less than len(class_name) in training?

### ISSUE THREAD

**Issue by ymzlygw** [qubvel/segmentation_models.pytorch#616]
Hi, Thanks for your work.
I used your code to train a multi-class segmentation task.  I have 10 classes, but after training, the output mask just have 8 classes( using np.unique).
Do you know how this happend? Thanks for your reply!

**Comment by Arunavameister** [comment]
I have a similar problem where my output has 3 classes whereas I expect 5. 
The training seems to be fine and the the IoU(during training) seems quite high as well.
@ymzlygw Have you found a solution yet?

**Comment by Arunavameister** [comment]
@qubvel 
Have you ever encountered this problem?

**Comment by qubvel** [comment]
Your model does not predict some classes, don't see anything strange here
suppose you have classes: car, road, sky, grass
but on the picture you have only: road, grass and sky

in that case if model not predict "car" and number of predicted classes is less than number of total classes, and that's correct

**Comment by Arunavameister** [comment]
Thanks for the reply, what you are saying is obviously true but in my case I am doing inference only on the **train set** and I have a really low loss and a really high IoU which makes me think that I am obviously missing something. 
The model should atleast overfit. I dont expect you to debug the problem for me ofcourse, was just wondering if this has been encountered before.
